packages/js-commons/src/local-files.ts

```diff
@@ -0,0 +1,380 @@
+    ) as ApplicationMetaDTO[];
+
+    const restoredData = data.map((app) => ({
+      ...app,
+      collaborators: new Map(Object.entries(app.collaborators ?? "")),
+    }));
+
+    return restoredData;
+  } catch (error) {
+    return [];
+  }
+};
+
+export const setAppsMetaData = async (
+  globalFolder: string,
+  defaultYamlPath: string,
+  appsMetaData: ApplicationMetaDTO[],
+): Promise<void> => {
+  // Write all apps manifest data in a single file
+  if (appsMetaData.length) {
+    let fileData: ApplicationMetaDTO[] = [];
+
+    ensureDir(globalFolder);
+    const filePath = join(globalFolder, AppManifestDataFile);
+
+    if (existsSync(filePath))
+      fileData = JSON.parse(
+        await readFile(filePath, "utf-8"),
+      ) as ApplicationMetaDTO[];
+
+    const localPathMap = new Map(
+      fileData.map((app) => [app.id, app.yamlFolderPath]),
+    );
+
+    // Convert the `collaborators` Map to an object for each app to store
+    const transformedApps = appsMetaData.map((metaData) => {
+      const yamlFolderPath = localPathMap.get(metaData.id);
+      return {
+        ...metaData,
+        yamlFolderPath: yamlFolderPath ?? join(defaultYamlPath, metaData.name), // Add path if there's a match or a default path
+        ...(metaData.collaborators && {
+          collaborators: Object.fromEntries(metaData.collaborators),
+        }),
+      };
+    });
+
+    await writeJsonData(filePath, transformedApps);
+  }
+};
+
+const readJsonFile = async <T>(filePath: string): Promise<T | undefined> => {
+  try {
+    if (!existsSync(filePath)) return undefined;
+
+    const fileContents = await readFile(filePath, "utf-8");
+    return JSON.parse(fileContents) as T;
+  } catch (error) {
+    return undefined;
+  }
+};
+
+const writeYamlFiles = async <T extends EnsembleDocument>(
+  folderPath: string,
+  artifact: T[],
+): Promise<void> => {
+  ensureDir(folderPath);
+
+  const artifactMetaData: Record<string, string> = {};
+  const filePromises = artifact.map(async (item) => {
+    const yamlFilePath = join(folderPath, `${item.name}.yaml`);
+    artifactMetaData[item.id] = item.name;
+    if (!existsSync(yamlFilePath))
+      await writeFile(yamlFilePath, item.content, "utf-8");
+  });
+
+  filePromises.push(
+    writeJsonData(join(folderPath, HiddenManifestFile), artifactMetaData, true),
+  );
+
+  await Promise.all(filePromises);
+};
+
+const writeYamlThemeFile = async <T extends EnsembleDocument>(
+  folderPath: string,
+  artifact: T,
+): Promise<void> => {
+  ensureDir(folderPath);
+  const yamlFilePath = join(folderPath, "Theme.yaml");
+  const artifactMetaData = { [artifact.id]: "Theme.yaml" };
+  if (!existsSync(yamlFilePath))
+    await Promise.all([
+      writeFile(yamlFilePath, artifact.content, "utf-8"),
+      writeJsonData(
+        join(folderPath, HiddenManifestFile),
+        artifactMetaData,
+        true,
+      ),
+    ]);
+};
+
+const getYamlFolderPath = async (
+  appId: string,
+  yamlFolder: string,
```

Comment:
What is this for? Since the app can be anywhere on the local FS I don't see the point of adding another path here?

Reply:
Since app can be anywhere on the FS this is the global metadata file to keep track of yaml folder on the fs.

Reply:
Then what is the difference between YAMLFolder and GlobalFolder?

Reply:
The idea was in the future users could update the path of App Yaml files from the web studio if they had moved the app from anywhere. This Yaml Folder path will act as a fallback path if they accidentally remove the Yaml Filepath to save the app data again.